src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,81 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const numberOfItems = Math.ceil(total / perPage);
+  const isSelectedFirst = currentPage === 1;
+  const isSelectedLast = currentPage === numberOfItems;
+
+  return (
+    <ul className="pagination">
+      <li className={classNames(
+        'page-item',
+        {
+          disabled: isSelectedFirst,
+        },
+      )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={isSelectedFirst}
+          onClick={() => onPageChange(currentPage - 1)}
+        >
+          «
+        </a>
+      </li>
+
+      {Array.from(Array(numberOfItems + 1).keys()).slice(1).map(num => (
```

Comment:
better move it to a separate variable and try to simplify this expression. you only need to push numbers diapason to an array